detectron2/modeling/meta_arch/rcnn.py

```diff
@@ -198,7 +199,7 @@ def _postprocess(instances, batched_inputs, image_sizes):
         ):
             height = input_per_image.get("height", image_size[0])
             width = input_per_image.get("width", image_size[1])
-            r = detector_postprocess(results_per_image, height, width)
+            r = retry_if_cuda_oom(detector_postprocess)(results_per_image, height, width)
```

Comment:
`results_per_image` is a structure that `retry_if_cuda_oom` probably can't handle.
It's better to use it for (or inside) `paste_masks_in_image`.

Reply:
`results_per_image` is an `Instance` so normally it should support `.to(device)`, but anyway it's as you've said better to use it for `paste_masks_in_image`

From what I've checked it's not a good idea to do it inside since there are different behaviors wether the device is `cpu` or `cuda` (and also the function checks the device of the input to decide)